### PR TITLE
fix(sdk-core): microgateway token generation

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Fix the way the microgateway token is calculated

### CODE REVIEW

File: packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts [hunk 1]
```diff
@@ -1,22 +1,45 @@
...
 
 /**
  * Computes the SHA256 digest of the given string
  * @param value Value to hash
  */
 export async function sha256(value: string) {
-  // Web browsers
   if (typeof window !== 'undefined' && typeof window.crypto !== 'undefined') {
-    return ab2str(await crypto.subtle.digest('SHA-256', str2ab(value)));
+    return sha256Browser(value);
```
Comment (divdavem) [inline packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts:38]: Depending on which node.js versions you target, maybe you do not really need two different versions for node.js and the browser. It looks like the browser version also works in node.js (v20.11.0).

File: packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
+}
+
+/**
+ *
+ * @param value
+ */
+export function sha256Node(value: string) {
+  const { createHash }: { createHash: typeof createHashType } = require('node:crypto');
+  return createHash('sha256')
+    .update(value, 'latin1')
```
Comment (divdavem) [inline packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts:28]: This does not match the browser version that uses utf8.

File: packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts (hunk 1 above)
Comment (kpanot) [reply packages/@ama-sdk/core/src/plugins/mgw-mdw-auth/mgw-mdw-auth.request.ts:38]: indeed, seems the case from Node 15 (cf. https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest)

**Comment by divdavem** [comment]
Actually, is this plugin intended to run in a browser environment?
I don't see how it can be useful without using the `createHmac` function from node:crypto.
I don't think it would be hard to make it work in a browser as well, though with the [crypto.subtle.sign function](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#hmac).

**Comment by kpanot** [comment]
> Actually, is this plugin intended to run in a browser environment? I don't see how it can be useful without using the `createHmac` function from node:crypto. However, I don't think it would be hard to make it work in a browser as well, with the [crypto.subtle.sign function](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#hmac).

No, it is not (their is a throw in the constructor if not used on not env) but the function `sha256` is exported and its previous implementation supported (theorically) both Browser and Node envs.
(I said "theorically" because it was actually not working correcly since few node versions)

**Review by fpaul-1A — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**